src/idl_parser.cpp

```diff
@@ -2757,4 +2857,95 @@ std::string Parser::ConformTo(const Parser &base) {
   return "";
 }
 
+// Locate all code related to LOCALE_INDEPENDENT in one compilation unit.
+// Is it better to return this code to the util.h?
+#if defined(FLATBUFFERS_FORCE_LOCALE_INDEPENDENT)
+// Static RAII holder of default C-locale
+class UtilDefaultLocale final {
+ public:
+  // clang-format off
+  #ifdef _MSC_VER
+  typedef _locale_t locale_type;
+  #else
+  typedef locale_t locale_type;
+  #endif
+  // clang-format on
+  static locale_type get() { return instance_.locale_; }
+
+ private:
+  static UtilDefaultLocale instance_;
+  locale_type locale_;
+
+  // clang-format offx
+  #ifdef _MSC_VER
+  UtilDefaultLocale() : locale_(_create_locale(LC_ALL, "C")) {}
+  ~UtilDefaultLocale() { _free_locale(locale_); }
+  #else
+  UtilDefaultLocale() : locale_(newlocale(LC_ALL, "C", (locale_t)0)) {}
+  ~UtilDefaultLocale() { freelocale(locale_); }
+  #endif
+  // clang-format onx
+};
+// allocate locate at startup of application
+UtilDefaultLocale UtilDefaultLocale::instance_;
+
+#endif
+
+double strtod_impl(const char *str, char **str_end){
+  // Result of strtod (printf, etc) depends from current C-locale.
+  // Most of locales use the dot '.' as decimal delimiter, but some use ','
+  // instead.
+  // User should predefine macro FLATBUFFERS_FORCE_LOCALE_INDEPENDENT to enable
+  // strtod_l/strtof_l usage.
```

Comment:
can we detect this automatically to not have to bother the user with it?

Reply:
With help of CMake we can detect that `strtod_l` exist in system (Linux) during build and automatically link it.
But should 99.9% of users pay for this? Most of them dont know about locales.
This could be actual if Flatbuffers library is part of another library or SDK (my case) and a developer isn't end-user.
Worst case - if Flatbuffers library is installed to `/usr/lib`.

I can write small performance test `strtod` vs `strtod_l` and check under Ubuntu and Windows to estimate cost of `strtod_l`.

Reply:
I found that `glibc` implements [strtod](https://github.com/lattera/glibc/blob/895ef79e04a953cac1493863bcae29ad85657ee1/stdlib/strtod.c) as to `strtod_l` with current locale pointer as the argument.
This means that we do not lose anything in performance (at least on Linux).

Reply:
Yeah, it sounds like there is always some locale involved.